pubsub/pubsub.go

```diff
@@ -0,0 +1,91 @@
+package pubsub
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/spf13/cobra"
+
+	"github.com/likecoin/likecoin-chain-tx-indexer/logger"
+)
+
+const (
+	CmdProjectID = "pubsub-project-id"
+	CmdTopic     = "pubsub-topic"
+)
+
+var (
+	client *pubsub.Client
+	topic  *pubsub.Topic
+	lock   = &sync.Mutex{}
+)
+
+func InitPubsubFromCmd(cmd *cobra.Command) error {
+	projectID, err := cmd.Flags().GetString(CmdProjectID)
+	if err != nil {
+		return err
+	}
+	topicID, err := cmd.Flags().GetString(CmdTopic)
+	if err != nil {
+		return err
+	}
+	if topicID == "" {
+		logger.L.Infow("Pubsub topic is empty, pubsub disabled")
+		return nil
+	}
+	logger.L.Infow("Pubsub enabled", "project_id", projectID, "topic", topicID)
+	return InitPubsub(projectID, topicID)
+}
+
+func InitPubsub(projectID, topicID string) error {
+	lock.Lock()
+	defer lock.Unlock()
+	if client == nil {
+		c, err := pubsub.NewClient(context.Background(), projectID)
+		if err != nil {
+			return err
+		}
+		client = c
+	}
+	if topic == nil {
+		topic = client.Topic(topicID)
+	}
+	return nil
+}
+
+type PubsubPayload struct {
+	Event   string      `json:"event"`
+	Payload interface{} `json:"payload"`
+}
+
+func Publish(action string, payload interface{}) error {
+	lock.Lock()
+	defer lock.Unlock()
+	if topic == nil {
+		return nil
+	}
+	encodedBz, err := json.Marshal(PubsubPayload{
+		Event:   "eventIndexer" + action,
+		Payload: payload,
```

Comment:
We will be using a pubsub topic just for chain events, so no need prefix for event
I suggest this schema
```suggestion
		Event:  action,
		Source: 'indexer',
		ChainId: '(add this from env?)',
		Payload: payload,
```